trace/exporterutil/exclude_spans.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exporterutil
+
+import (
+	"go.opencensus.io/trace"
+	"strings"
+)
+
+// SpanMatcher allows matching SpanData by well-defined criteria.
+type SpanMatcher struct {
+	// we deliberately keep all SpanMatcher implementations known in this
+	// package so it's easier for us to define them in configuration in future.
+	matches func(*trace.SpanData) (shouldExport bool)
+}
+
+// WithNamePrefix returns a SpanMatcher that matches only Spans with names
+// starting with the given prefix.
+// Note: some exporters add a "Sent." or "Recv." prefix to the exported
+// span name at export time (based on the SpanKind). This filter does not
+// apply to these SpanKind-prefixed names, but rather the original name of the
+// span.
+func WithNamePrefix(prefix string) SpanMatcher {
+	return SpanMatcher{
+		matches: func(spanData *trace.SpanData) bool {
+			return strings.HasPrefix(spanData.Name, prefix)
+		},
+	}
+}
+
+// WithSpanKind matches spans with SpanKind equal to the supplied value.
+func WithSpanKind(kind int) SpanMatcher {
+	return SpanMatcher{
+		matches: func(spanData *trace.SpanData) bool {
+			return spanData.SpanKind == kind
+		},
+	}
+}
+
+// ExcludingSpans wraps an exporter and drops any spans that match any one or
+// more of the filters.
+func ExcludingSpans(e trace.Exporter, excludeIfAllMatch ...SpanMatcher) *ExcludingSpansExporter {
+	return &ExcludingSpansExporter{e, excludeIfAllMatch}
+}
+
+type ExcludingSpansExporter struct {
+	e       trace.Exporter
+	exclude []SpanMatcher
+}
+
+var _ ExportFlusher = (*ExcludingSpansExporter)(nil)
+
+func (e *ExcludingSpansExporter) ExportSpan(spanData *trace.SpanData) {
+	exclude := true
+	for _, matcher := range e.exclude {
+		if !matcher.matches(spanData) {
+			exclude = false
+			break
+		}
+	}
+	if !exclude {
+		e.e.ExportSpan(spanData)
+	}
+}
+
+func (e *ExcludingSpansExporter) Flush() {
```

Comment:
What is Flush?

Reply:
Flush is a convention implemented by Stackdriver and (I think) other exporters that do internal buffering.